### PR TITLE
Increase size of column 'value' at table 'account_details'

### PR DESCRIPTION
**Pull request by GabrielBrascher**
### Description

This PR increases the column `value` at table `account_details` from 255 chars to `4096`, matching with the value allowed in the API command for updating the configuration of accounts.

When the `value` length is bigger than 255, the following log is presented right after the `updateConfiguration` API call:
```
2022-03-09 17:50:24,627 ERROR [c.c.a.ApiServer] (qtp30578394-234766:ctx-cad18b45 ctx-32e954dd) (logid:0948e203) unhandled exception executing api command: [Ljava.lang.String;@117c6ba7
com.cloud.utils.exception.CloudRuntimeException: DB Exception on: com.mysql.cj.jdbc.ClientPreparedStatement: INSERT INTO account_details (account_details.account_id, account_details.name, account_details.value) VALUES (123, _binary'api.allowed.source.cidr.list', _binary'<huge binary>')
	at com.cloud.utils.db.GenericDaoBase.persist(GenericDaoBase.java:1450)
	at jdk.internal.reflect.GeneratedMethodAccessor168.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	....
	....
	....
Caused by: com.mysql.cj.jdbc.exceptions.MysqlDataTruncation: Data truncation: Data too long for column 'value' at row 1
	at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:104)
	at com.mysql.cj.jdbc.ClientPreparedStatement.executeInternal(ClientPreparedStatement.java:953)
	at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1092)
	... 83 more
```

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [X] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [X] Minor
- [ ] Trivial

### How Has This Been Tested?

- Update configurations for an account
- Assert that configuration is properly updated (which was not the case before the change)

### CODE REVIEW

**Comment by GabrielBrascher** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test

**Review by nvazquez — APPROVED** [review]
Code LGTM

**Review by davidjumani — APPROVED** [review]

**Review by Pearl1594 — APPROVED** [review]

**Comment by GabrielBrascher** [comment]
Before merging, I am afraid that we might need to also increase the size of the value in the response object, otherwise, the API response will not contain the whole value stored in DB. Placing it as a draft in the meantime to avoid it getting merged.

**Comment by nvazquez** [comment]
Thanks @GabrielBrascher I was spinning an env for testing it - will post my review after that

**Comment by GabrielBrascher** [comment]
Thanks, @nvazquez!
I will do some changes and run some manual tests.

**Comment by nvazquez** [comment]
@GabrielBrascher I've tested through the API and UI:

- API works as expected, passed `value` parameters with length > 255 and the response was properly retrieved. 
- UI seems to still be truncating the value at 255 characters, can you check if this can be extended?

Example: input length = 650 characters:

````
(localcloud) 🐱 > update configuration accountid=deb22344-0869-49a2-ac1f-116cb858ee03 name='api.allowed.source.cidr.list' value=12345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890
{
  "configuration": {
    "category": "Advanced",
    "description": "Comma separated list of IPv4/IPv6 CIDRs from which API calls can be performed. Can be set on Global and Account levels.",
    "isdynamic": true,
    "name": "api.allowed.source.cidr.list",
    "scope": "account",
    "value": "12345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890"
  }
}
````
Screen Shot 2022-03-10 at 10 13 45

**Comment by GabrielBrascher** [comment]
@nvazquez  I am seeing the following:

- Update command and response looks OK:

```
(poc) 🐱 > update configuration accountid=064a02be-caf8-46c7-8264-92c5a9d782b2 name=api.allowed.source.cidr.list value=12345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890
{
  "configuration": {
    "category": "Advanced",
    "description": "Comma separated list of IPv4/IPv6 CIDRs from which API calls can be performed. Can be set on Global and Account levels.",
    "isdynamic": true,
    "name": "api.allowed.source.cidr.list",
    "scope": "account",
    "value": "12345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890"
  }
}
```

- But list command returns a fraction:

```
(poc) 🐱 > list configurations name=api.allowed.source.cidr.list accountid=064a02be-caf8-46c7-8264-92c5a9d782b2 pagesize=600 page=1
{
  "configuration": [
    {
      "category": "Advanced",
      "description": "Comma separated list of IPv4/IPv6 CIDRs from which API calls can be performed. Can be set on Global and Account levels.",
      "isdynamic": true,
      "name": "api.allowed.source.cidr.list",
      "scope": "account",
      "value": "123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345"
    }
  ],
  "count": 1
}
```

Updated value with length of 651 

```
"value": "12345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890"
```

listed value with length of 256
```
"value": "123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345"
```

**Comment by nvazquez** [comment]
Yes @GabrielBrascher seeing the same - the only remark from your comment is the lengths are 650 and 255. Extending the response field should solve it, worth checking if the UI does truncate or simply displays the received data for configurations

**Comment by nvazquez** [comment]
Hi @GabrielBrascher please advise when this one is ready for testing again

File: engine/schema/src/main/resources/META-INF/db/schema-41610to41700.sql [hunk 1]
```diff
@@ -652,3 +652,5 @@ INSERT INTO `cloud`.`user_vm_details`(`vm_id`, `name`, `value`)
         INNER JOIN `cloud`.`vm_instance` ON vm_instance.id = user_vm_details.vm_id
         WHERE ssh_keypairs.account_id = vm_instance.account_id;
 
+-- PR #6080 Change column `value` size from 255 to 4096 characters, matching the API "updateConfiguration" "value" size
+ALTER TABLE `cloud`.`account_details` MODIFY `value` VARCHAR(4096);
```
Comment (GutoVeronezi) [inline engine/schema/src/main/resources/META-INF/db/schema-41610to41700.sql:656]: When calling `MODIFY` command, it removes all the column metadata and only adds what is specified in the command (in this case, only `VARCHAR(4096)`. Currently this field is `NOT NULL`, therefore it must be informed in the command:

```suggestion
ALTER TABLE `cloud`.`account_details` MODIFY `value` VARCHAR(4096) NOT NULL;
```

**Review by GutoVeronezi — CHANGES_REQUESTED** [review]

File: engine/schema/src/main/resources/META-INF/db/schema-41610to41700.sql (hunk 1 above)
Comment (GabrielBrascher) [reply engine/schema/src/main/resources/META-INF/db/schema-41610to41700.sql:656]: Thanks @GutoVeronezi!

**Comment by weizhouapache** [comment]
@GabrielBrascher 
can you address the comments and fix the issue in your testing ?

**Comment by shwstppr** [comment]
@blueorangutan package

File: engine/schema/src/main/resources/META-INF/db/schema-41610to41700.sql [hunk 2]
```diff
@@ -652,3 +652,5 @@ INSERT INTO `cloud`.`user_vm_details`(`vm_id`, `name`, `value`)
         INNER JOIN `cloud`.`vm_instance` ON vm_instance.id = user_vm_details.vm_id
         WHERE ssh_keypairs.account_id = vm_instance.account_id;
 
+-- PR #6080 Change column `value` size from 255 to 4096 characters, matching the API "updateConfiguration" "value" size
+ALTER TABLE `cloud`.`account_details` MODIFY `value` VARCHAR(4096) NOT NULL;
```
Comment (shwstppr) [inline engine/schema/src/main/resources/META-INF/db/schema-41610to41700.sql:656]: @BartJM @GabrielBrascher can you please move the schema change to 4.17.1 upgrade path

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@BartJM I asked you to move the schema change to 4.17.1 upgrade path as milestone is set to 4.17.1 but the PR is targeted for main, do you want it to go only into main? If you want it in 4.17.1 can you please rebase and target it to 4.17 branch. Else you will need to again move schema changes into 41710to41800 upgrade path. Sorry for not mentioning it earlier.
cc @GabrielBrascher

File: engine/schema/src/main/resources/META-INF/db/schema-41700to41710.sql [hunk 3]
```diff
@@ -126,3 +126,6 @@ CREATE VIEW `cloud`.`domain_router_view` AS
         `cloud`.`async_job` ON async_job.instance_id = vm_instance.id
             and async_job.instance_type = 'DomainRouter'
             and async_job.job_status = 0;
+
+-- PR #6080 Change column `value` size from 255 to 4096 characters, matching the API "updateConfiguration" "value" size
+ALTER TABLE `cloud`.`account_details` MODIFY `value` VARCHAR(4096) NOT NULL;
```
Comment (Rubueno) [inline engine/schema/src/main/resources/META-INF/db/schema-41700to41710.sql:131]: @BartJM Can you ensure a newline is present?

**Comment by BartJM** [comment]
@GabrielBrascher could you move the target branch to 4.17? I am unable to do so.

**Comment by GabrielBrascher** [comment]
@BartJM done.

**Comment by shwstppr** [comment]
@blueorangutan package

**Review by shwstppr — APPROVED** [review]

**Comment by shwstppr** [comment]
@blueorangutan test

**Pull request merged**